qupulse/expressions.py

```diff
@@ -240,6 +240,11 @@ def __str__(self) -> str:
     def __repr__(self) -> str:
         return 'Expression({})'.format(repr(self._original_expression))
 
+    def __format__(self, format_spec):
+        if format_spec=='':
+            return str(self)
+        return f'%{format_spec}' % float(self)
```

Comment:
Can't we just replace this with
```python
return format(float(self), format_spec)
```

Reply:
Seems good, I didn't know about the `format` method.